toqito/state_props/is_separable.py

```diff
@@ -148,11 +176,40 @@ def is_separable(state: np.ndarray, dim: None | int | list[int] = None, level: i
     # Check these tests.
     if min_dim == 2:
         # Check if X is separable from spectrum.
-        if (lam[0] - lam[2 * max_dim - 1]) ** 2 <= 4 * lam[2 * max_dim - 2] * lam[2 * max_dim] + tol**2:
+        if (lam[0] - lam[2 * max_dim - 2]) ** 2 <= 4 * lam[2 * max_dim - 3] * lam[2 * max_dim - 1] + tol**2:
```

Comment:
Commenting here since I think there was a minor mistake in the indexing before this PR.

As far as I see, `array[i]` in `numpy` is the same as `array(i+1)` in MATLAB.

Reply:
It's possible there was an issue before. The existing unit tests never covered this specific line.

Reply:
> ... this line lacks a test, so putting a case in to cover it would be great

For the same reason as other places where I could not add a test and as discussed previously, it might be helpful to think about a more principled approach to design the test cases but leave those tests for now.

Reply:
Okay, sure. I will create an issue about the uncovered lines for which tests must be added if and once this PR is nearly merged!

Reply:
@Yash-10 There is already an issue for uncovered lines in `is_separable`. It might be better to add a comment in #293